### PR TITLE
[JENKINS-72018] Form entries in .repeated-chunk using .show-if-not-only CSS class are not hidden when they are alone

### PR DESCRIPTION
**Pull request by flabrie**
Form entries in .repeated-chunk using .show-if-not-only CSS class are not hidden when they are alone, because there are missing CSS class definitions.

See [JENKINS-72018](https://issues.jenkins.io/browse/JENKINS-72018).

### Testing done

Yes, and adding the missing CSS classes fix the issue.

### Proposed changelog entries

- Hide the delete button from the only repeatable element in configuration forms when at least one element is expected. (regression in 2.344)

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [X] There is automated testing or an explanation as to why this change has no tests.
- [X] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [X] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [X] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [X] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [X] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8491"></a>

### CODE REVIEW

**Comment by daniel-beck** [comment]
Effectively reverts https://github.com/jenkinsci/jenkins/pull/6471.

Probably re-introduces the warnings motivating that PR (but for correct behavior, seems the lesser evil).

**Comment by daniel-beck** [comment]
Interactive test with Subversion plugin (see screenshots in Jira) passes, but autotests fail.

Unsure how to go about fixing those, perhaps JTH change to make https://github.com/jenkinsci/jenkins-test-harness/blob/3b9790f39c3106398edda1133833caf9c31532af/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L2337 more tolerant, reverting parts of https://github.com/jenkinsci/jenkins-test-harness/pull/530? Looks like HTMLUnit may lack support for what's a legal selector.

**Comment by daniel-beck** [comment]
Per https://github.com/HtmlUnit/htmlunit-cssparser/issues/36, `only` seems to be treated like a reserved word in HTMLUnit's CSS parser.

Awkward that it's documented in https://github.com/jenkinsci/jenkins/blob/c5fbbbe88dc280d0a0876614c0f9aa6fba2ea2af/core/src/main/resources/lib/form/repeatable.jelly#L54 so just renaming the class might be out.

Perhaps
```
$ git diff war/src/
diff --git a/war/src/main/scss/form/_reorderable-list.scss b/war/src/main/scss/form/_reorderable-list.scss
index dfd0fc5c5a..c19152975d 100644
--- a/war/src/main/scss/form/_reorderable-list.scss
+++ b/war/src/main/scss/form/_reorderable-list.scss
@@ -29,7 +29,7 @@
   visibility: visible;
 }
 
-.repeated-chunk.only .show-if-not-only {
+.repeated-chunk.first.last .show-if-not-only {
   visibility: hidden;
 }
 
@@ -54,7 +54,7 @@
   visibility: visible;
 }
 
-.repeated-chunk .repeated-chunk.only .show-if-not-only {
+.repeated-chunk .repeated-chunk.first.last .show-if-not-only {
   visibility: hidden;
 }
 
@@ -79,7 +79,7 @@
   visibility: visible;
 }
 
-.repeated-chunk .repeated-chunk .repeated-chunk.only .show-if-not-only {
+.repeated-chunk .repeated-chunk .repeated-chunk.first.last .show-if-not-only {
   visibility: hidden;
 }
```

This makes the autotest pass at least (not manually tested).

File: war/src/main/scss/form/_reorderable-list.scss
Comment (daniel-beck) [inline war/src/main/scss/form/_reorderable-list.scss:32]:
```suggestion
.repeated-chunk.first.last .show-if-not-only {
```

File: war/src/main/scss/form/_reorderable-list.scss
Comment (daniel-beck) [inline war/src/main/scss/form/_reorderable-list.scss:57]:
```suggestion
.repeated-chunk .repeated-chunk.first.last .show-if-not-only {
```

File: war/src/main/scss/form/_reorderable-list.scss
Comment (daniel-beck) [inline war/src/main/scss/form/_reorderable-list.scss:82]:
```suggestion
.repeated-chunk .repeated-chunk .repeated-chunk.first.last .show-if-not-only {
```

**Review by daniel-beck — CHANGES_REQUESTED** [review]
(Per previous comment)

**Review by daniel-beck — APPROVED** [review]
Works well in interactive testing with Subversion SCM modules in freestyle jobs.

Nesting of selectors looks error-prone, but this basically only restores what we had before the "cleanup", so seems tolerable.

When going from "not only" to "only", the "X" disappears well before the grey circle, but it doesn't look too broken.

**Review by timja — APPROVED** [review]
nice fix

----

/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
